core/src/smartcontracts/isi/triggers/set.rs

```diff
@@ -539,6 +540,42 @@ pub enum LoadedExecutable {
     Instructions(Vec<InstructionBox>),
 }
 
+impl<F: Clone> From<&Action<F, LoadedExecutable>> for SerialAction<F> {
+    fn from(r: &Action<F, LoadedExecutable>) -> Self {
+        let exec = match &r.executable {
+            LoadedExecutable::Wasm(lw) => Executable::Wasm(WasmSmartContract(
+                lw.module
+                    .serialize()
+                    .expect("This failing means iroha was compiled incorrectly"),
+            )),
+            LoadedExecutable::Instructions(vec) => Executable::Instructions(vec.clone()),
+        };
+        SerialAction {
+            executable: exec,
+            repeats: r.repeats.clone(),
+            authority: r.authority.clone(),
+            filter: r.filter.clone(),
+            metadata: r.metadata.clone(),
+        }
+    }
+}
+
+#[derive(Encode, Decode)]
+pub struct SerialAction<F> {
```

Comment:
This struct looks exactly the same as `data_model::trigger::Action`, no?

Reply:
`data_model::trigger::Action<F, Executable>` to be precise

Reply:
They are not semantically the same. So I'm fine with some duplication

Reply:
On that note, I'd like this struct to be defined before it's being used. Could you move it above `impl From`?